Website/Providers/DataProviders/SqlDataProvider/08.01.00.01.SqlDataProvider

```diff
@@ -166,3 +166,65 @@ AS
 
 	EXEC {databaseOwner}{objectQualifier}EnsureLocalizationExists @PortalID = @PortalId, @CultureCode = @CultureCode
 GO
+
+-- DNN-8358: Improve Journal_Save to support updates
+IF EXISTS (SELECT * FROM sys.Procedures WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}Journal_Save]'))
+    DROP PROCEDURE {databaseOwner}[{objectQualifier}Journal_Save]
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}Journal_Save]
+    @PortalID           Int,
+    @JournalId          Int,
+    @JournalTypeId      Int,
+    @UserID             Int,
+    @ProfileId          Int,
+    @GroupId            Int,
+    @Title              nVarChar(255),
+    @Summary            nVarChar(2000),
+    @ItemData           nVarChar(2000),
+    @JournalXML         xml,
+    @ObjectKey          nVarChar(255),
+    @AccessKey          UniqueIdentifier,
+    @SecuritySet        nVarChar(2000),
+    @CommentsDisabled   Bit,
+    @CommentsHidden     Bit
+AS
+BEGIN
+    IF IsNull(@JournalId, -1) = -1 BEGIN
```

Comment:
DNN never uses transactions, usually it uses single statements per procedure.
The Journal is one of the most Anti-Database pieces in the structure - storing relations compacted in a single text column is just against all concepts. There are no foreign keys defined and I don't really see a reason to create a dedicated table for a single xml column.
My fix here is just to solve this one issue - because Active Forums module needs to delete and recreate a journal item upon every update, which is not reliable.